front/app/components/ConsentManager/ConsentModal.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { FormEvent } from 'react';
+
+// styling
+import styled from 'styled-components';
+
+// components
+import Link from 'utils/cl-router/Link';
+import Button from 'components/UI/Button';
+
+// i18n
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from './messages';
+import {
+  Box,
+  Text,
+  Title,
+  useBreakpoint,
+} from '@citizenlab/cl2-component-library';
+import Modal from 'components/UI/Modal';
+import Preferences from './Preferences/Preferences';
+import { CategorizedDestinations, IPreferences } from './typings';
+import { TCategory } from 'components/ConsentManager/destinations';
+import Footer from './Preferences/Footer';
+import ContentContainer from './Preferences/ContentContainer';
+
+const StyledLink = styled(Link)`
+  text-decoration: underline;
+
+  &:hover,
+  &:focus {
+    text-decoration: underline;
+  }
+`;
+
+interface Props {
+  onAccept: () => void;
+  onClose: () => void;
+  showModal: boolean;
+  setShowModal: (showModal: boolean) => void;
+  categorizedDestinations: CategorizedDestinations;
+  updatePreference: (category: TCategory, value: boolean) => void;
+  preferences: IPreferences;
+  handleCancelBack: () => void;
+  handleCancelConfirm: () => void;
+  handleCancel: () => void;
+  handleSave: (e: FormEvent) => void;
+  mode: 'preferenceForm' | 'noDestinations' | 'cancelling';
+  isCancelling: boolean;
+  view: 'main' | 'preferences';
+  setView: (view: 'main' | 'preferences') => void;
+}
+
+const ConsentModal = ({
+  onAccept,
+  onClose,
+  showModal,
+  categorizedDestinations,
+  updatePreference,
+  preferences,
+  mode,
+  isCancelling,
+  handleCancelBack,
+  handleCancelConfirm,
+  handleCancel,
+  handleSave,
+  view,
+  setView,
+}: Props) => {
+  const isPhoneOrSmaller = useBreakpoint('phone');
+
+  return (
+    <Modal
```

Comment:
We can probably separate out some code here for the "main" view vs "preferences" view. Though the Modal component itself will still have to conditionally render the title/footer depending on the view (if we want everything to render in the same modal).